### PR TITLE
feat(content-separator): allow ContentSeparator in Menus

### PR DESCRIPTION
**Pull request by dominiccarrington**
# Description

Update \<ContentSeparator\> to be classed as a valid component in \<Menu\>, \<MenuSection\> and \<MenuSelectionGroup\>

### CODE REVIEW

File: src/components/Menu/Menu.utils.test.tsx [hunk 1]
```diff
@@ -10,55 +10,100 @@ jest.mock('@react-stately/collections', () => ({
...
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore - We don't export this method in the types
+    const generator = MenuSeperator.getCollectionNode(props);
+    const results = Array.from(generator);
+
+    expect(results).toHaveLength(1);
+    expect(results[0]).toEqual({
+      type: 'seperator',
+      props: expect.any(Object),
```
Comment (nataliadelmar) [inline src/components/Menu/Menu.utils.test.tsx:103]: Could we do: 
    expect(results[0]).toEqual({
      type: 'seperator',
      props: props,
    });

And avoid the next 2 lines below?

File: src/components/Menu/Menu.unit.test.tsx [hunk 2]
```diff
@@ -344,6 +360,38 @@ describe('<Menu />', () => {
...
+      await user.keyboard('{ArrowDown}');
+
+      expect(menuItems[0]).toHaveFocus();
+
+      await user.keyboard('{ArrowRight}');
+
+      expect(menuItems[0]).toHaveFocus();
+    });
+
     it('should handle up/down arrow keys correctly - for vertical menu with section', async () => {
```
Comment (nataliadelmar) [inline src/components/Menu/Menu.unit.test.tsx:395]: would be also great to add MenuSeparator in this test

Comment (nataliadelmar) [reply src/components/Menu/Menu.unit.test.tsx:395]: as well as in the selectiongroup test below

File: src/components/Menu/Menu.stories.tsx [hunk 3]
```diff
@@ -256,14 +225,10 @@ SelectionGroups.parameters = {
...
           title={
-            <>
-              <ListHeader outline={true} outlinePosition="top" outlineColor="secondary">
-                <ListItemBaseSection position="start">
-                  <Icon scale={16} name="accessibility" strokeColor="none" />
-                </ListItemBaseSection>
-                <ListItemBaseSection position="fill">Layout</ListItemBaseSection>
-              </ListHeader>
-            </>
+            <Flex direction="row" alignItems="center" xgap="0.25rem">
```
Comment (nataliadelmar) [inline src/components/Menu/Menu.stories.tsx:228]: nothing to do with your PR, but could you remove the aria-labels from all selectionGroups on this story? They all have a visible title that labels the group, so the aria-label is lost anyways and it is confusing to have it there

**Review by nataliadelmar — COMMENTED** [review]
lgtm! only some non-blocking comments :D

File: src/components/Menu/Menu.utils.test.tsx (hunk 1 above)
Comment (dominiccarrington) [reply src/components/Menu/Menu.utils.test.tsx]: Simplified the test

File: src/components/Menu/Menu.unit.test.tsx (hunk 2 above)
Comment (dominiccarrington) [reply src/components/Menu/Menu.unit.test.tsx:395]: Done!

**Review by nataliadelmar — APPROVED** [review]

File: src/components/Menu/Menu.utils.test.tsx [hunk 4]
```diff
@@ -1,64 +1,101 @@
...
+    resultNodes.forEach((node, index) => {
+      expect(node).toEqual({
+        ...mockNodes[index],
+        props: { ...selectionGroupProps, selectionGroup: true },
       });
     });
+  });
+});
+
+describe('ContentSeparator', () => {
```
Comment (shnaaz) [inline src/components/Menu/Menu.utils.test.tsx:66]: I think this test should live in ContentSeparator.unit.test.tsx

Comment (dominiccarrington) [reply src/components/Menu/Menu.utils.test.tsx]: Moved

File: src/components/Menu/Menu.utils.test.tsx [hunk 5]
```diff
@@ -1,64 +1,64 @@
 import React from 'react';
 import { SelectionGroup } from './Menu.utils';
-import { Item , Section, PartialNode } from '@react-stately/collections';
-import {  SelectionMode } from '@react-types/shared';
+import { Item, Section, PartialNode } from '@react-stately/collections';
+import { SelectionMode } from '@react-types/shared';
 import { SelectionGroupProps } from './Menu.types';
+import ContentSeparator from '../ContentSeparator';
```
Comment (nataliadelmar) [inline src/components/Menu/Menu.utils.test.tsx:6]: do we need this import for something?

File: src/components/MenuSelectionGroup/MenuSelectionGroup.style.scss [hunk 6]
```diff
@@ -10,4 +10,8 @@
   font-size: 0.875rem;
   line-height: 1.375rem;
   padding: 0.25rem 0.75rem;
+
+  .md-text-wrapper {
+    font-size: 0.875rem;
```
Comment (nataliadelmar) [inline src/components/MenuSelectionGroup/MenuSelectionGroup.style.scss:15]: are we using the Text component anywhere in the header?
font-size should normally be set on the Text component as type="subheader/title/etc"
I'd prefer not setting it in css

File: src/components/Menu/Menu.utils.test.tsx (hunk 5 above)
Comment (dominiccarrington) [reply src/components/Menu/Menu.utils.test.tsx:6]: Not anymore

File: src/components/MenuSelectionGroup/MenuSelectionGroup.style.scss (hunk 6 above)
Comment (dominiccarrington) [reply src/components/MenuSelectionGroup/MenuSelectionGroup.style.scss:15]: We are using the Text component in the SelectionGroup stories. The same font-size can be achieved with type="body-secondary". I have reverted this change.

**Comment by shnaaz** [comment]
Tested with Cantina, works fine

**Review by shnaaz — APPROVED** [review]

**Pull request merged**